iceoryx_utils/include/iceoryx_utils/internal/cxx/expected.inl

```diff
@@ -115,7 +115,7 @@ inline expected<ValueType, ErrorType> expected<ValueType, ErrorType>::create_err
 template <typename ValueType, typename ErrorType>
 inline expected<ValueType, ErrorType>::operator bool() const noexcept
 {
-    return m_hasError;
+    return !m_hasError;
```

Comment:
there is another `operator bool` in the `expected<ErrorType>` specialization. Could you also fix that one.
Would also be nice if you write a test for it.

... and since you are already at it, there are some deprecated methods which could be removed, like `on_success`. @elfenpiff what do you think?

Reply:
This is a great idea. Let's get rid of them!

Reply:
Cool stuff. Get this PR merged as I just needed `operator bool` in port_manager.cpp and roudi_process.cpp in #360!

Reply:
Sure. We should add a test for `expected<ErrorType>`, thoug. Currently they only cover `expected<ValueType, ErrorType>`